junit-jupiter-engine/src/main/java/org/junit/jupiter/engine/descriptor/MethodBasedTestDescriptor.java

```diff
@@ -77,10 +79,6 @@ public final Method getTestMethod() {
 
 	@Override
 	public String getLegacyReportingName() {
-		return generateDefaultDisplayName(testMethod);
-	}
-
-	private static String generateDefaultDisplayName(Method testMethod) {
 		return String.format("%s(%s)", testMethod.getName(),
```

Comment:
Shouldn't this simply delegate to the `DEFAULT` style in order to adhere to the DRY principle?

Reply:
Good catch!

Reply:
Erm, no! The line belongs to `getLegacyReportingName`, which should not be changed. Or should it? 🤔

Reply:
This method implements `org.junit.platform.engine.TestDescriptor#getLegacyReportingName` -- delegating to `DEFAULT` seems odd here.

Reply:
Why? That's what it used to do.... delegate to the default.

_You_ changed it so that the code is now duplicated. 😇

Reply:
Right. We could handle it like that.

But at the same time, you make a valid point: it's just a coincidence that they are the same.

Thus, if we delegate to the _default_, we should at least make sure we have a "regression test" in place for the legacy one to ensure that it doesn't accidentally diverge in the future (if we change the _default_).

Reply:
Actually, we should have a regression test in place in any case. 😉

Reply:
Regarding violation of the DRY principle, I'm leaning toward your original idea, that it's safer to duplicate for these two use cases.

I'll leave it up to you... 😇

Reply:
...let's keep the code duplication.

Reply:
SGTM!

Care to introduce the regression test to go along with it? 😉

Reply:
_und Neo so:_ 😵